packages/data-helper/src/hooks/useNodeInfo/useNodeInfo.gql-queries.js

```diff
@@ -100,14 +104,15 @@ const getBaseQueryAndVariables = variables => {
         generatedVariables: {
             path: variables.path
         },
-        skip: !variables.path || variables.path.length === 0
+        skip: !variables.path || variables.path.length === 0,
+        supportsExistsInLive: variables.supportsExistsInLive
     };
 };
 
 export const getQuery = (variables, options = {}) => {
     const fragments = [];
 
-    const {baseQuery, generatedVariables, skip} = getBaseQueryAndVariables(variables);
+    const {baseQuery, generatedVariables, skip, supportsExistsInLive} = getBaseQueryAndVariables(variables);
```

Comment:
no need to modify getBaseQueryAndVariables, you can take the schema directly from the getQuery parameters. getBaseQueryAndVariables do not use it at all. 
schema can be checked directly in `if (options.getAggregatedPublicationInfo)` block, when needed. 
Loading state can be handled here - if schema is not loaded, you can return a new "loading" flag at the end that will be handled by useNodeInfo().

Reply:
yes